dicetable.py

```diff
@@ -0,0 +1,327 @@
+            for x in range(self.dsize):
+                
+                newdic[el+x+1] = (newdic.get(el+x+1,0)+currentval)
+        self.table= newdic
+     
+    def __str__(self):
+        return str(self.totaldice)+'D'+str(self.dsize)    
+    def printTable(self):
+        '''go on. give it a try.  you'll never guess what this function does.
+        it's a surprise'''
+        for el in range(self.min(), self.max()+1):
+            if el<10:
+                print ' '+str(el)+':'+str(self.table[el])
+            else:
+                print(str(el)+':'+str(self.table[el]))  
+                
+    def mean(self):
+        '''i mean, don't you just sometimes look at a table of values
+        and wonder what the mean is?'''
+        return (self.totaldice*(1+self.dsize)/2.0) 
+    def stddev(self):
+        '''wassamatter you! you don't know what standard deviation is?'''
+        try:
+            avg = self.mean()
+            sqs = 0
+            count = 0
+            for x in self.table.keys():
+                sqs += self.table[x]*((avg - x)**2)
+                count += self.table[x]
+            return round((sqs/count)**0.5,4)
+        except OverflowError:
+            print 'this standard deviation is a gross approximation '+\
+            'as i had to int() all the sqrt, because the numbers are just too big'
+            avg = self.mean()
+            sqs = 0
+            count = 0
+            for x in self.table.keys():
+                sqs += self.table[x]*int((avg - x)**2)
+                count += self.table[x]
+            return round((sqs/count)**0.5,4)
+        
+    def grapher(self):
+        '''returns a graph of self'''
+        maxx = self.table[int(self.mean())]           
+        divisor = 1
+        divstring = '1'
+        #this sets the divisor so that max height of graph is 80 x's
+        if maxx > 80 and maxx<=80000000:
+            divisor = maxx/80.0
+            divstring = str(divisor)
+        if maxx > 80000000:
+            #this code just outputs as a string with sci notation assuming no decimals
+            divisor = maxx/80
+            divstring = str(divisor)
+            power = len(divstring)-1
+            digits = divstring[0]+'.'+divstring[1:4]
+            divstring =digits+'e+'+str(power)
+        
+                
+        for x in range (self.min() , self.max()+1):
+            val = self.table.get(x,0)
+            if x<10:
+                print ' '+str(x)+': '+(int(round(val/divisor)))*'x'
+            else:
+                print str(x)+': '+(int(round(val/divisor)))*'x'
+        print 'each x represents '+(divstring)+' occurences'      
+        print self
+
+    def truncategrapher(self):
+        '''prints a graph of self with  the 0-'x' bits removed'''
+        maxx = self.table[int(self.mean())]           
+        divisor = 1
+        divstring = '1'
+        
+        if maxx > 80 and maxx<=80000000:
+            divisor = maxx/80.0
+            divstring = str(divisor)
```

Comment:
actually since those values are used multiple times in the function they should be  extracted to local variables.